hack/verify-govet.sh

```diff
@@ -30,6 +30,62 @@ do
   fi
 done
 
+# For the sake of slowly white-listing `shadow` checks, we need to keep track of which
+# directories we're searching through. The following are all of the directories we care about:
+# all top-level directories except for 'pkg', and all second-level subdirectories of 'pkg'.
+ALL_DIRS=$(find_files | grep -Eo "\./([^/]+|pkg/[^/]+)" | sort -u)
+
+DIR_BLACKLIST='./assets
+./cmd
+./doc.go
+./examples
+./hack
+./images
+./pkg/api
+./pkg/assets
+./pkg/auth
+./pkg/authorization
+./pkg/build
+./pkg/client
+./pkg/cmd
+./pkg/config
+./pkg/controller
+./pkg/deploy
+./pkg/diagnostics
+./pkg/dns
+./pkg/doc.go
+./pkg/dockerregistry
+./pkg/generate
+./pkg/gitserver
+./pkg/image
+./pkg/ipfailover
+./pkg/oauth
+./pkg/project
+./pkg/route
+./pkg/router
+./pkg/sdn
+./pkg/security
+./pkg/service
+./pkg/serviceaccounts
+./pkg/template
+./pkg/user
+./pkg/util
+./pkg/version
+./plugins
+./test'
+
+for test_dir in $ALL_DIRS
+do
+  # use `grep` failure to determine that a directory is not in the blacklist 
+  if ! echo "${DIR_BLACKLIST}" | grep -q "${test_dir}"; then
+    go tool vet -shadow -shadowstrict $test_dir
+    if [ "$?" -ne 0 ]
```

Comment:
seems odd to compare a string and a number... this probably works, but make both sides one or the other